tests/unit/CMakeLists.txt

```
@@ -32,7 +32,7 @@ function(add_unit_test MAIN_SOURCE)
   target_compile_definitions(${BASENAME} PRIVATE ${UNIT_COMPILE_DEFINITIONS})
   add_symlink_to_built_opencl_dynlib(${BASENAME})
 
-  target_link_libraries(${BASENAME} PRIVATE ${POCL_LIBRARY_NAME})
+  target_link_libraries(${BASENAME} PRIVATE ${POCL_LIBRARY_NAME} ${POCLU_LINK_OPTIONS})
```

(copied from the previous PR)

It would be better if the link options were not passed in via `target_link_libraries` due to reasons in [CMake docs](https://cmake.org/cmake/help/latest/command/target_link_libraries.html) (generally considered unsafe, placement of the options may not be correct).
```suggestion
  target_link_options(${BASENAME} PRIVATE ${POCL_LIBRARY_NAME} ${POCLU_LINK_OPTIONS})
```

---

The name is a misnomer. `POCLU_LINK_OPTIONS` includes libraries as well.

---

> POCLU_LINK_OPTIONS includes libraries as well.

I see, there could be cmake targets in it AFAICS - can `target_link_options` consume them?

Anyway, I'll give this a pass.

---

No it cannot consume targets.